pandas/tests/io/formats/test_to_html.py

```diff
@@ -415,6 +415,91 @@ def test_to_html_multiindex_max_cols(self, datapath):
+        df.index = MultiIndex.from_product([['a', 'b'],
+                                            ['c', 'd', 'e', 'f', 'g']])
+        result = df.to_html(index=False)
+        expected = expected_html(datapath, 'gh22579_expected_output')
+        assert result == expected
+
+    @pytest.mark.parametrize('idx_flag, named_index, expected_output', [
+        (True, True, 'gh22579_truncation_index_true'),
+        (False, True, 'gh22579_truncation_index_false_named'),
+        (False, False, 'gh22579_truncation_index_false_unnamed')
+    ])
+    def test_to_html_multi_indexes_index_false_with_truncation(
+            self, datapath, idx_flag, named_index, expected_output):
+        # GH 22579 with truncation
+        names = ['foo', None, 'baz'] if named_index else None
+        index = MultiIndex.from_product([['a', 'b'], ['c', 'd'], ['e', 'f']],
+                                        names=names)
+        df = DataFrame(np.arange(64).reshape(8, 8), index=index, columns=index)
+        result = df.to_html(max_rows=4, max_cols=4, index=idx_flag)
+        expected = expected_html(datapath, expected_output)
+        assert result == expected
+
+    @pytest.mark.parametrize(
+        'index_names', [True, False], ids=['index_names_true',
+                                           'index_names_false'])
+    @pytest.mark.parametrize(
+        'index', [True, False], ids=['index_true', 'index_false'])
+    @pytest.mark.parametrize('col_idx_type', [
+        (Index([0, 1]),
+         'columns_unnamed_standard'),
+        (Index([0, 1], name='columns.name'),
+         'columns_named_standard'),
+        (MultiIndex.from_product([['a'], ['b', 'c']]),
+         'columns_unnamed_multi'),
+        (MultiIndex.from_product([['a'], ['b', 'c']],
+                                 names=['columns.name.0',
+                                        'columns.name.1']),
+         'columns_named_multi')
+    ],
+        ids=['columns_unnamed_standard',
+             'columns_named_standard',
+             'columns_unnamed_multi',
+             'columns_named_multi'])
+    @pytest.mark.parametrize('idx_type', [
+        (Index([0, 1]),
+         'index_unnamed_standard'),
+        (Index([0, 1], name='index.name'),
+         'index_named_standard'),
+        (MultiIndex.from_product([['a'], ['b', 'c']]),
+         'index_unnamed_multi'),
+        (MultiIndex.from_product([['a'], ['b', 'c']],
+                                 names=['index.name.0',
+                                        'index.name.1']),
+         'index_named_multi')
+    ],
+        ids=['index_unnamed_standard',
+             'index_named_standard',
+             'index_unnamed_multi',
+             'index_named_multi'])
+    def test_to_html_basic_alignment(self, datapath, idx_type, col_idx_type,
+                                     index, index_names):
+        # GH 22747, GH 22579
+        idx_type, idx_type_id = idx_type
```

Comment:
> the only methods i've found on getting the id inside the test function all appear a bit hacky. the 'id' seems to be mainly to make the test output more readable.

> If inspecting the id isn't an option couldn't you just pass in the name as an argument instead?

in response to your comment the ids are passed along with the index.. using a tuple is a hack to pass the id along with the index, they are not 2 separate parameters. introspecting the indexes was a better solution in my opinion.

if i pass just the id as a parameter then the index would need to be constructed within the test. with header parametrization we create 256 indexes instead of 8 and even more if we add additional tests when a fixture could have been used.

Reply:
since the indexes have been moved from a fixture into the parametrize decorator for this PR, then with

`@pytest.mark.parametrize('col_idx_type, col_idx_type_id', [
        (Index([0, 1]),
         'columns_unnamed_standard'),`

`def test_to_html_basic_alignment(
            self, datapath, idx_type, idx_type_id, col_idx_type,
            col_idx_type_id, index, index_names):`

it is possible for this PR to remove the unpacking but this would prevent easily making it a fixture in the future.

Reply:
a side-effect of adding the type as an extra parameter is the test log file is readable again.. nice!

pandas/tests/io/formats/test_to_html.py::TestToHTML::test_to_html_basic_alignment[row_index1-named_standard-column_index3-named_multi-False-False] PASSED [ 66%]

Reply:
following this PR i was hoping to extend the parametric alignment checking to the other output formatting methods. hence much refactoring here was to create fixtures separate from the actual test which could have easily been copy and pasted to conftest.py in the next PR. we'll cross that bridge...